### PR TITLE
initial Docker setup

### PR DESCRIPTION
**Pull request by dylanvaughn**
@nlamprian can you try:

    $ make compile
    $ make bringup

and let me know if it works for you?  You'll need `Docker`, `docker-compose`, and `make` installed locally.

### CODE REVIEW

**Comment by nlamprian** [comment]
It compiles mostly fine, but the bringup fails

```
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
libGL error: No matching fbConfigs or visuals found
libGL error: failed to load driver: swrast
[ WARN] [1612081365.819676832]: OGRE EXCEPTION(3:RenderingAPIException): Unable to create a suitable GLXContext in GLXContext::GLXContext at /build/ogre-1.9-kiU5_5/ogre-1.9-1.9.0+dfsg1/RenderSystems/GL/src/GLX/OgreGLXContext.cpp (line 61)
rviz::RenderSystem: error creating render window: OGRE EXCEPTION(3:RenderingAPIException): Unable to create a suitable GLXContext in GLXContext::GLXContext at /build/ogre-1.9-kiU5_5/ogre-1.9-1.9.0+dfsg1/RenderSystems/GL/src/GLX/OgreGLXContext.cpp (line 61)
[ERROR] [1612081365.864385110]: Unable to create the rendering window after 100 tries.
terminate called after throwing an instance of 'std::logic_error'
  what():  basic_string::_M_construct null not valid
libGL error: No matching fbConfigs or visuals found
libGL error: failed to load driver: swrast
X Error of failed request:  GLXBadContext
  Major opcode of failed request:  151 (GLX)
  Minor opcode of failed request:  6 (X_GLXIsDirect)
  Serial number of failed request:  38
  Current serial number in output stream:  37
```

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        python3-pip                           \
+        python3-apt                           \
+        python3-vcstool                       \
+        python3-colcon-common-extensions      \
+        && rm -rf /var/lib/apt/lists/*
+RUN rosdep update
+
+# colcon
+RUN pip3 install -U setuptools
+RUN pip3 install colcon-ros-bundle
```
Comment (nlamprian) [inline Dockerfile:34]: You can probably replace this with `python3-colcon-ros`

Comment (nlamprian) [reply Dockerfile:34]: Actually, it's already a dependency of python3-colcon-common-extensions. You may wanna just try removing this line.

**Comment by dylanvaughn** [comment]
@nlamprian can you try these and see if it fixes things for you?

https://askubuntu.com/a/1211225

https://github.com/jessfraz/dockerfiles/issues/253

**Comment by nlamprian** [comment]
Indeed, I just needed to install the nvidia driver

**Comment by dylanvaughn** [comment]
I'm going to redo this PR since the directory rename is giving me some issues...new PR to come shortly.

**Pull request closed without merging**